packages/gestalt/src/TableRow.js

```diff
@@ -1,10 +1,52 @@
 // @flow strict
 import * as React from 'react';
+import styles from './Table.css';
 
 type Props = {|
   children: React.Node,
+  onClick?: ({
+    event:
+      | SyntheticMouseEvent<HTMLDivElement>
+      | SyntheticKeyboardEvent<HTMLDivElement>,
+  }) => void,
 |};
 
+const SPACE_CHAR_CODE = 32;
+const ENTER_CHAR_CODE = 13;
+
 export default function TableRow(props: Props) {
-  return <tr>{props.children}</tr>;
+  const { children, onClick } = props;
+
+  const handleKeyPress = (event: SyntheticKeyboardEvent<HTMLDivElement>) => {
```

Comment:
@PayalModi could you update with `HTMLDivElement` here and below with `HTMLTableRowElement` since we add these handlers to a `tr`